cpacs2to3/material.py

```diff
@@ -0,0 +1,437 @@
+                              ("thermalConductivity31", self.thermalConductivity[2]),
+                              ("thermalExpansionCoeff12", self.thermalExpansionCoeff[1]),
+                              ("thermalExpansionCoeff23", self.thermalExpansionCoeff[4]),
+                              ("thermalExpansionCoeff31", self.thermalExpansionCoeff[2]),]
+
+        if not tixi.checkElement(xPath):
+            tixi.createElement(*xPath.rsplit('/',1))
+        for name, value in namesAndValues:
+            if value is not None and abs(value)>1e-8:
+                # value must be not None and != zero
+                tixi.addTextElement(xPath, name, str(value))
+
+    def setStrength(self, strength):
+        """This method is intended to set the strength of the material."""
+        self.strength.update(strength)
+
+    def removeCpacs2Definitions(self, tixi):
+        """removes the cpacs2 definitions in the material node at self.xPath"""
+        elemsToRemove = ["fatigueFactor",
+                         "thermalConductivity",
+                         "k11", "k12", "k13", "k22", "k23", "k33", "k44", "k55", "k66",
+                         "sig11",
+                         "tau12",
+                         "sig11yieldT",
+                         "sig11yieldC",
+                         "sig11t",
+                         "sig11c",
+                         "sig22t",
+                         "sig22c",
+                         "sig33t",
+                         "sig33c",
+                         "tau12",
+                         "tau23",
+                         "tau13",
+                         "sig33t",
+                         "sig33c",
+                         "maxStrain",
+                         "postFailure",
+                         ]
+        for element in elemsToRemove:
+            xPath = self.xPath + '/' + element
+            if tixi.checkElement(xPath):
+                tixi.removeElement(xPath)
+
+
+    def _getModuli(self):
+        """
+        calculates moduli
+
+        :return: dict with these keys: e11, e22, e33, g12, g23, g13, nu12, nu21, nu31, nu31, nu23
+        """
+        if self._savedModuli != {}:
+            return self._savedModuli
+        stiffnessM = self.stiffnessMatrix
+        try:
+            complianceM = np.linalg.inv(stiffnessM)
+        except np.linalg.LinAlgError:
+            raise ValueError(
+                "Please check your material definition! "
+                + "Could not calculate compliance matrix of material element at xPath "
+                + self.xPath
+            )
+
+        e11 = complianceM[0, 0] ** -1
+        e22 = complianceM[1, 1] ** -1
+        e33 = complianceM[2, 2] ** -1
+
+        g23 = complianceM[3, 3] ** -1
+        g13 = complianceM[4, 4] ** -1
+        g12 = complianceM[5, 5] ** -1
+
+        nu12 = -complianceM[1, 0] * e11
+        nu21 = -complianceM[0, 1] * e22
+        nu31 = -complianceM[0, 2] * e33
```

Comment:
Line 391 and 393 are both setting nu31. This looks like an error!

Reply:
yep, adjusted